File: Day 115/classwork/cw1.py
# 1) https://www.codewars.com/kata/523f5d21c841566fde000009/train/python


# 1


def array_diff(a, b):
    res = []
    for i in a:
        if i not in b:
            res.append(i)
    return res


#  2
array_diff = lambda a, b:  [x for x in a if x not in b]

# 3

def array_diff(a, b):
    return [x for x in a if x not in b]

File: Day 165/classwork/cw1.js
// 1) https://www.codewars.com/kata/56582133c932d8239900002e/train/javascript


function mostFrequentItemCount(collection) {
    let maxCount = 0;
    let counts = {};

    for(let i = 0; i < collection.length; i++){
        let num = collection[i];

        counts[num] = (counts[num] || 0) + 1;
        if(counts[num] > maxCount){
            maxCount = counts[num];
        }
    }
    return maxCount
}

console.log(mostFrequentItemCount([2, 4, 5, 4, 5, 3, 4, 4, 4]))